packages/hooks/use-posts-infinite-scroll.tsx

```diff
@@ -0,0 +1,242 @@
+import React, { useEffect } from "react";
+import { connect, css, useConnect } from "frontity";
+import { Connect } from "frontity/types";
+import useInfiniteScroll from "./use-infinite-scroll";
+import WpSource from "@frontity/wp-source/types";
+import TinyRouter from "@frontity/tiny-router/types";
+
+type Wrapper = React.FC<Connect<WpSource & TinyRouter, { link: string }>>;
+
+type UsePostsInfiniteScroll = (options: {
+  limit?: number;
+  context?: string;
+}) => {
+  posts: {
+    key: string;
+    link: string;
+    isLastPost: boolean;
+    Wrapper: Wrapper;
+  }[];
+  isLimit: boolean;
+  isFetching: boolean;
+  fetchNext: () => Promise<void>;
+};
+
+const Wrapper: Wrapper = connect(({ state, link, children }) => {
+  // Values from browser state.
+  const links: string[] = state.router.state.links || [link];
+  const limit: number = state.router.state.limit;
+  const pages: string[] = state.router.state.pages || [];
+
+  // Aliases to needed state.
+  const current = state.source.get(link);
+  const first = links[0];
+  const items = pages.reduce((final, current, index) => {
+    const data = state.source.get(current);
+    if (data.isArchive && data.isReady) {
+      const items =
+        index !== 0
+          ? data.items.filter(({ link }) => link !== first)
+          : data.items;
+      final = final.concat(items);
+    }
+    return final;
+  }, []);
+  const currentIndex = items.findIndex(({ link }) => link === current.link);
+  const nextItem = items[currentIndex + 1];
+
+  // Infinite scroll booleans.
+  const hasReachedLimit = !!limit && links.length >= limit;
+
+  const { supported, fetchRef, routeRef } = useInfiniteScroll({
+    currentLink: link,
+    nextLink: nextItem?.link,
+  });
+
+  if (!current.isReady) return null;
+  if (!supported) return children;
+
+  const container = css`
+    position: relative;
+  `;
+
+  const fetcher = css`
+    position: absolute;
+    width: 100%;
+    bottom: 0;
+  `;
+
+  return (
+    <div css={container} ref={routeRef}>
+      {children}
+      {!hasReachedLimit && <div css={fetcher} ref={fetchRef} />}
+    </div>
+  );
+});
+
+const usePostsInfiniteScroll: UsePostsInfiniteScroll = (options) => {
```

Comment:
Could you please rename this to `usePostTypeInfiniteScroll` (singular)? I want to start using the same terms everywhere, and we are going to start using `postType` for custom post types, including posts and pages. Thanks!